src/Netsells/SSOClient/Pomerium.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Netsells\SSOClient;
+
+use Firebase\JWT\JWT;
+use Illuminate\Http\Request;
+
+class Pomerium
+{
+    protected static $publicKey;
+
+    protected const HEADER_NAME = "X-Pomerium-Jwt-Assertion";
+
+    protected $userId;
+    protected $email;
+    protected $firstName;
+    protected $lastName;
+    protected $fullName;
+    protected $photoUrl;
+    protected $groups = [];
+
+    public static function setPublicKey($key): void
+    {
+        static::$publicKey = base64_decode($key);
+    }
+
+    public static function fromRequest(Request $request): Pomerium
+    {
+        $pomerium = new static();
+
+        if ($headerValue = $request->header(Pomerium::HEADER_NAME)) {
+            $pomerium->decodeHeader($headerValue);
+        }
+
+        return $pomerium;
+    }
+
+    private function __construct()
+    {
+
+    }
```

Comment:
Could just remove this as it's unused

Reply:
This is intentional actually - a trick @kubatek94 showed me, setting it to private prevents you from doing a standard 'new up'

Reply:
🤯